protected/humhub/modules/user/components/UsernameValidator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2018 HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\user\components;
+
+use Yii;
+use yii\validators\Validator;
+
+/**
+ * UsernameValidator checks that username contains only lowercase letters, numbers, hyphens and underscores
+ *
+ */
+class UsernameValidator extends Validator
```

Comment:
Currently this validator is only used when changing the password, but not in the User Model itself, furthermore the pattern and message should be configurable e.g. by a [Module Class Level Configuration](http://docs.humhub.org/dev-modules-base-class.html#module-class-level-configuration)